### PR TITLE
fix(component_mesh_polyhedra): Fixed issue when trying to get polyhedra adjacent to an internal surface of the block.

### PR DESCRIPTION
**Pull request by MelchiorSchuh**

### CODE REVIEW

File: src/geode/model/helpers/component_mesh_polyhedra.cpp [hunk 1]
```diff
@@ -35,50 +35,100 @@
...
+    class PolyhedronVerticesPossibilities
+    {
+    public:
+        PolyhedronVerticesPossibilities(
             absl::Span< const std::vector< geode::ComponentMeshVertex > >
                 unique_vertices_cmvs )
-    {
-        if( unique_vertices_cmvs.empty() || unique_vertices_cmvs[0].empty() )
+            : unique_vertices_cmvs_( unique_vertices_cmvs ),
+              nb_unique_vertices_{ static_cast< geode::local_index_t >(
```
Comment (BotellaA) [inline src/geode/model/helpers/component_mesh_polyhedra.cpp:45]: Should we use a local_index_t? Nothing in the design ensure this

File: src/geode/model/helpers/component_mesh_polyhedra.cpp [hunk 2]
```diff
@@ -35,50 +35,100 @@
...
         }
-        const auto nb_unique_vertices = unique_vertices_cmvs.size();
+
         std::vector< std::pair< geode::uuid, geode::PolyhedronVertices > >
-            common_block_vertices_list;
-        for( const auto& first_cmv : unique_vertices_cmvs[0] )
+            compute_possibilities()
+        {
+            if( unique_vertices_cmvs_.empty()
+                || unique_vertices_cmvs_[0].empty() )
```
Comment (BotellaA) [inline src/geode/model/helpers/component_mesh_polyhedra.cpp:54]: Why only check for [0] and not for all the unique vertices?

File: src/geode/model/helpers/component_mesh_polyhedra.cpp [hunk 3]
```diff
@@ -35,50 +35,100 @@
...
+                const auto& mesh_vertices = block_mesh_vertices( first_cmv );
+                if( mesh_vertices.empty() )
+                {
+                    continue;
+                }
+                geode::PolyhedronVertices current_list;
+                current_list.push_back( first_cmv.vertex );
+                fill_polyhedronvertices_possibilities(
+                    first_cmv.component_id.id(), mesh_vertices, 1,
+                    std::move( current_list ) );
```
Comment (BotellaA) [inline src/geode/model/helpers/component_mesh_polyhedra.cpp:70]:
```suggestion
                fill_polyhedronvertices_possibilities(
                    first_cmv.component_id.id(), mesh_vertices, 1,
                    { first_cmv.vertex } );
```
maybe try this?

File: src/geode/model/helpers/component_mesh_polyhedra.cpp [hunk 4]
```diff
@@ -35,50 +35,100 @@
...
-                    break;
+                    return {};
                 }
-                mesh_vertices.push_back( it->vertex );
             }
-            if( mesh_vertices.size() == nb_unique_vertices )
+            return mesh_vertices;
+        }
+
+        void fill_polyhedronvertices_possibilities(
```
Comment (panquez) [inline src/geode/model/helpers/component_mesh_polyhedra.cpp:104]: I would name it "fill_polyhedron_vertices_possibilities"

**Review by panquez — APPROVED** [review]

File: src/geode/model/helpers/component_mesh_polyhedra.cpp (hunk 1 above)
Comment (MelchiorSchuh) [reply src/geode/model/helpers/component_mesh_polyhedra.cpp:45]: We usually use local_index_t when dealing with the vertices of a polyhedron, but I can change it to index_t

**Review by BotellaA — APPROVED** [review]

**Pull request merged**